Exclude timing

alehaa opened the pull request:
Removed timing library, because it is now build independent as [LLNL-hires-timers](https://github.com/LLNL/llnl-hires-timers).

Additionally added LICENSE to all files and a (yet empty) README file and fixed some CMake bugs.

alehaa committed:
removed timing stuff

removed timing library and linked to external library LLNL
hires-timers, which is in fact the same but as external library.

alehaa committed:
added travis configuration file

alehaa committed:
Added libboost-all-dev to .travis.yml

alehaa committed:
Merge branch 'travis_ci' into exclude_timing

alehaa committed:
Added install section to .travis.yml to build LLNL hires-timers before adept-utils

alehaa committed:
Added LICENSE and README

- added LICENSE.md
- added README.md
- added license header to all files or replaced the old one

alehaa committed:
choose new behaviour for CMake CMP0042

alehaa committed:
Moved CMake commands into files were they are actually needed

alehaa committed:
fixed #2

Removed call for abs in utils/matrix_utils.h for unsigned values

alehaa committed:
Switched cmake build type in .travis.yml

alehaa committed:
Beautified root CMakeLists.txt

alehaa committed:
switched boost dependency in .travis.yml

jprotze commented:
@tgamblin do you consider acepting this PR, or should we drop the request?

tgamblin reviewed (changes requested):
@alehaa @jprotze: sorry for the delay.  I think this looks good.

Minor change request: could you *not* add the entire license to each file, and instead say something like "`adept-utils` is under a BSD 3-clause license, see the top-level LICENSE file for details"?

Also I've given you both commit access to this repository, as you are using it more than I am.  I think it makes sense for you to be able to maintain the library.

tgamblin commented:
Also @jprotze: can I trouble you for a response on spack/spack#9137?

alehaa committed:
Fix copyright header

Instead of prepending the complete LICENSE file in each file, only a
minimal license header will be used.

alehaa committed:
Update Travis CI config
